### ISSUE TITLE
Questions on basic documentation in README

### ISSUE THREAD

**Issue by dune73** [dextermallo/GSoC-2023#16]
As you are currently polishing the documentation, I have a few remarks where you could improve the basic overview, namely the architecture graph.

* You may want to define the term "matrix/matrices".
* The arrow from *Server-side Matrices* joins with "Client-side Matrices" and hits "Unified Data Format". Why does not this data run through the "Data Parser"?
* You have two blocks "Server-side Matrices". I think it makes sense to keep them separate in a logical view. But they should have distinct names, the way it stands, people will miss them up.
* I do not understand why one of the "Server-side Matrices" is only "Near Real-time" why it's client counterpart is "Real-time". I might have an idea, but I think you need to explain this in the document.
* The blue item "Client" should probably be "Client Collector"
* There is an arrow with desciption "partial requests". What does "partial" mean in this context?
* "Fuzz: fuzzing payloads and sending them to the WAF. e.g.," Your rule probably does not do what you want it to do (-> it blocks GET!). Please fix it and then write the rule with at least id, phase, disruptive action, log and msg. This makes mistakes much more obvious.
* "For instance, in this case, if we use POST instead of GET, , it will become a false positive". I do not understand this. Do you mean to block POST with the rule? If yes, then why is it a false positive and not a true positive?

This is probably not complete, but these things sprung to mind during a first detailed pass.

**Comment by dexter-ys** [comment]
Hello @dune73, I reviewed the doc again and revised it in the PR [#9](https://github.com/dextermallo/GSoC-2023/tree/feat/v2-performance-framework). 
I think most of the content is outdated 😅

With the revision, I think the only one I have to work on is: You may want to define the term "matrix/matrices".
Could you please review it again in the PR? Thank you!

**Comment by dune73** [comment]
There is no PR #9. Which one do you mean?

**Comment by dexter-ys** [comment]
> There is no PR #9. Which one do you mean?

Sorry, it should be #10

**Comment by dune73** [comment]
Thanks.

This looks good. I read through it and it makes sense. 

Am I right, you removed the diagram? I think you should create a diagram. The setup is hard to understand otherwise.

The quick start quite and the more extensive guide are very good. Decent example on top.

Please discuss the results in the graph in more detail. Look through it with the eyes of a newbie. What do we see? What numbers are uninteresting? Which numbers are important? Where do you see a change exactly? This will take some practice and you can give people a head start here.

**Comment by dexter-ys** [comment]
> Am I right, you removed the diagram? I think you should create a diagram. The setup is hard to understand otherwise.

Yes. Just removed it because I think the arch has changed a lot. I will draw a new one

> Please discuss the results in the graph in more detail. Look through it with the eyes of a newbie. What do we see? What numbers are uninteresting? Which numbers are important? Where do you see a change exactly? This will take some practice and you can give people a head start here.

Yah this makes sense. I will update this part.

Thanks for the feedback! 👍
